tests/ModelTestStateTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace RonasIT\Support\Tests;
+
+use ReflectionClass;
+use RonasIT\Support\Tests\Support\Mock\TestModel;
+use RonasIT\Support\Tests\Support\Traits\ModelTestStateMockTrait;
+use RonasIT\Support\Tests\Support\Traits\MockTrait;
+
+class ModelTestStateTest extends HelpersTestCase
+{
+    use ModelTestStateMockTrait;
+    use MockTrait;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        self::$tables = null;
+
+        putenv('FAIL_EXPORT_JSON=false');
+    }
+
+    public function testInitialization()
+    {
+        $datasetMock = collect($this->getJsonFixture('initialization/dataset.json'));
+        $originRecords = collect($this->getJsonFixture('initialization/origin_records.json'));
+
+        $this->mockGettingDataset($datasetMock);
+
+        $modelTestState = new ModelTestState(TestModel::class);
+
+        $reflectionClass = new ReflectionClass($modelTestState);
+        $jsonFieldsProperty = $reflectionClass->getProperty('jsonFields');
+        $jsonFieldsProperty->setAccessible(true);
+
+        $jsonFields = $jsonFieldsProperty->getValue($modelTestState);
```

Comment:
let's move it to the separate helper `getProtectedField($class, string $propertyName)`